network/ocsp_test.go

```diff
@@ -256,6 +259,80 @@ func TestDefaultOCSPClient(t *testing.T) {
 	checkCase(t, revokedData)
 }
 
+func testWithConfigAndValidChain(t *testing.T, ocspConfig *OCSPConfig, rawCerts [][]byte, verifiedChains [][]*x509.Certificate) {
+	ocspClient := DefaultOCSPClient{}
+
+	ocspVerifier := DefaultOCSPVerifier{Config: *ocspConfig, Client: ocspClient}
+
+	err := ocspVerifier.Verify(rawCerts, verifiedChains)
+	if err != nil {
+		t.Fatalf("Unexpected error for valid certificate: %v", err)
+	}
+}
+
+func testWithConfigAndRevokedChain(t *testing.T, ocspConfig *OCSPConfig, rawCerts [][]byte, verifiedChains [][]*x509.Certificate) {
+	ocspClient := DefaultOCSPClient{}
+
+	ocspVerifier := DefaultOCSPVerifier{Config: *ocspConfig, Client: ocspClient}
+
+	err := ocspVerifier.Verify(rawCerts, verifiedChains)
+	if err == nil {
+		t.Fatal("Unexpected success when verifying revoked certificate")
+	}
+	if !strings.Contains(err.Error(), "revoked") {
+		t.Logf("Verify error: %v", err)
+		t.Fatalf("Error does not contain 'revoked'")
+	}
+	t.Logf("(Expected) verify error: %v", err)
+}
+
 func TestDefaultOCSPVerifier(t *testing.T) {
-	// TODO
+	goodData := getOCSPTestCase(t, "../tests/ssl/acra-writer/acra-writer", ocsp.Good)
+	revokedData := getOCSPTestCase(t, "../tests/ssl/acra-writer-revoked/acra-writer-revoked", ocsp.Revoked)
+	// TODO switch CA cert -> OCSP responder cert as soon as OCSPVerifier can handle that
+	ocspCertificate, ocspSigningKey := getTestOCSPCertAndKey(t, "../tests/ssl/ca/ca")
+
+	ocspServerConfig := ocspServerConfig{
+		issuerCert:    goodData.issuer,
+		responderCert: ocspCertificate,
+		responderKey:  ocspSigningKey,
+		testCases:     []*ocspTestCase{goodData, revokedData},
+	}
+
+	ocspServer, addr := getTestOCSPServer(t, ocspServerConfig)
+	defer ocspServer.Close()
+
+	uri := fmt.Sprintf("http://%s", addr)
+
+	// TODO generate these chains by reading files, like getOCSPTestCase() does
+	validRawCerts, validVerifiedChains := getValidTestChain(t)
+	invalidRawCerts, invalidVerifiedChains := getInvalidTestChain(t)
+
+	//
+	// Test with default config, certificates contain OCSP server inside
+	//
+	ocspConfig, err := NewOCSPConfig("", ocspRequiredYesStr, ocspFromCertUseStr, false)
```

Comment:
better to split it to 3 tests even if we will need to copy paste some code. it will simplify to refactor separate cases and it will not touch other cases. plus it will simplify understanding where error occurred, when output will be something like:
```
TestDefaultOCSPVerifierValidChain  Success
TestDefaultOCSPVerifierInvalidChain Failure
TestDefaultOCSPVerifierEmptyOCSPServer Failure
```
instead `TestDefaultOCSPVerifier Failure`